x-pack/test/performance/journeys/base.config.ts

```diff
@@ -23,6 +24,18 @@ export default async function ({ readConfigFile, log }: FtrConfigProviderContext
 
   log.info(` 👷‍♀️ BUILD ID ${testBuildId}\n 👷 JOB ID ${testJobId}\n 👷‍♂️ EXECUTION ID:${executionId}`);
 
+  const telemetryLabels: TelemetryConfigLabels = {
+    branch: process.env.BUILDKITE_BRANCH,
+    ciBuildId: process.env.BUILDKITE_BUILD_ID,
+    ciBuildJobId: process.env.BUILDKITE_JOB_ID,
+    ciBuildNumber: Number(process.env.BUILDKITE_BUILD_NUMBER) || 0,
+    gitRev: process.env.BUILDKITE_COMMIT,
+    isPr: !!process.env.GITHUB_PR_NUMBER,
+    ...(process.env.GITHUB_PR_NUMBER ? { prId: process.env.GITHUB_PR_NUMBER } : {}),
+    testJobId,
+    testBuildId,
+  };
```

Comment:
Sorry, didn't think about this but since this object is being JSON encoded this will set `prId` to a string, we should parse it to number

```suggestion
  const prId = process.env.GITHUB_PR_NUMBER ? Number.parseInt(process.env.GITHUB_PR_NUMBER, 10) : undefined;
  
  if (Number.isNaN(prId)) {
    throw new Error('invalid GITHUB_PR_NUMBER environment variable');
  }
  
  const telemetryLabels: TelemetryConfigLabels = {
    branch: process.env.BUILDKITE_BRANCH,
    ciBuildId: process.env.BUILDKITE_BUILD_ID,
    ciBuildJobId: process.env.BUILDKITE_JOB_ID,
    ciBuildNumber: Number(process.env.BUILDKITE_BUILD_NUMBER) || 0,
    gitRev: process.env.BUILDKITE_COMMIT,
    isPr: prId !== undefined,
    ...(prId !== undefined ? { prId } : {}),
    testJobId,
    testBuildId,
  };
```

Reply:
`...(process.env.GITHUB_PR_NUMBER ? { prId: Number(process.env.GITHUB_PR_NUMBER) } : {})` wouldn't this be sufficient change ? 
GITHUB_PR_NUMBER is either string or undefined:
- if it's undefined we will dest. empty object which is basically we are not gonna have this property
- if it's string then it means it's not empty and has some value "I' assume", I also assume that string will always a string-number

Reply:
Possibly, but passing values to the `Number()` constructor does insufficient parsing for different number formats and there's always a chance that `GITHUB_PR_NUMBER` is set to something we don't expect, maybe because someone tweaks the value or because something in Buildkite has changed. Adding parsing validation always seems like a good idea to me.

Reply:
Sure sounds good to me then 👍🏽